Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeInstance.java

```diff
@@ -210,6 +228,24 @@ public Long getFileObjectId() {
         return objectId;
     }
 
+    /**
+     * Get the accountId of the account associated with the correlation
+     * attribute.
+     *
+     * @return the accountId of the account
+     */
+    public Long getAccountId() {
+        return accountId;
+    }
+
+    /**
+     * Set the accountId of the account associated with this correlation
+     * attribute.
+     */
+    public void setAccountId(Long accountId) {
```

Comment:
Is the comment about why the setter is public? Or why there is a setter at all?

Many of the Correlation attributes do not have any account.  And depending upon the usage of the CorrelationAttributeInstance (for e.g. in the Other instance viewer in main Autopsy window) the backing account is not needed at all.   I'm keeping the Id instead of the account itself so that it can be read where needed and is not unnecessarily preloaded all the time.

Reply:
I'm cool with this. Just to clarify my comment was questioning the setter at all and then my next proposal was making the signature:

public void setAccount(CentralRepoAccount account)

I should've used signature and not abstraction. But anyway, it's a slightly stricter contract. We know this attributeinstance can only have an accountId that is stored in the DB.

In its current state, Long is too broad. The range of negatives it supports isn't logically valid for a CentralRepoAccount, but of course it'll compile. That's all.